controllers/nicclusterpolicy_controller_test.go

```diff
@@ -0,0 +1,141 @@
+/*
+2023 NVIDIA CORPORATION & AFFILIATES
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers //nolint:dupl
+
+import (
+	goctx "context"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	mellanoxv1alpha1 "github.com/Mellanox/network-operator/api/v1alpha1"
+	"github.com/Mellanox/network-operator/pkg/consts"
+)
+
+//nolint:dupl
+var _ = Describe("NicClusterPolicyReconciler Controller", func() {
+
+	Context("When NicClusterPolicy CR is created", func() {
+		It("should create and delete whereabout", func() {
```

Comment:
nit: can you elaborate a bit more on the It statement 

as this test:
1. creates nicclusterpolcy CR with ipam plugin
2. ensure objects created
3. update nicclusterpolicy CR with nil val for secondary netwokr
4. ensure related  obj deleted

smthn like: should create whereabouts and delete it after un-setting CR value

Reply:
Done